### PR TITLE
Bridge & CacheBridge with relay multiple

### PR DESCRIPTION
**Pull request by prin-r**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: bridges/evm/contracts/CacheBridge.sol [hunk 1]
```diff
@@ -60,4 +60,29 @@ contract CacheBridge is Bridge, ICacheBridge {
 
         requestsCache[requestKey] = res;
     }
+
+    /// Performs oracle state relay and many times of oracle data verification in one go.
+    /// After that, the results which is an array of Packet will be recorded to the state by using the hash of RequestPacket as key.
+    /// @param _data The encoded data for oracle state relay and an array of data verification.
+    function relayMulti(bytes calldata _data) external {
```
Comment (taobun) [inline bridges/evm/contracts/CacheBridge.sol]: Add this function in `ICacheBridge` interface

File: bridges/evm/contracts/ProxyBridge.sol [hunk 2]
```diff
@@ -48,14 +47,21 @@ contract ProxyBridge is IBridge, ICacheBridge, Ownable {
...
+        return bridge.relayAndMultiVerify(_data);
+    }
+
+    /// Performs oracle state relay and oracle data verification in one go.
+    /// After that, the results will be recorded to the state by using the hash of RequestPacket as key.
+    /// @param _data The encoded data for oracle state relay and data verification.
+    function relay(bytes calldata _data) external override {
         return bridge.relay(_data);
     }
 }
```
Comment (taobun) [inline bridges/evm/contracts/ProxyBridge.sol:74]: relayMulti?

**Review by taobun — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/CacheBridge.sol (hunk 1 above)
Comment (prin-r) [reply bridges/evm/contracts/CacheBridge.sol]: done

File: bridges/evm/contracts/ProxyBridge.sol (hunk 2 above)
Comment (prin-r) [reply bridges/evm/contracts/ProxyBridge.sol:74]: done

File: bridges/evm/contracts/CacheBridge.sol [hunk 3]
```diff
@@ -60,4 +60,29 @@ contract CacheBridge is Bridge, ICacheBridge {
...
+    /// After that, the results which is an array of Packet will be recorded to the state by using the hash of RequestPacket as key.
+    /// @param _data The encoded data for oracle state relay and an array of data verification.
+    function relayMulti(bytes calldata _data) external override {
+        Packet[] memory packets = this.relayAndMultiVerify(_data);
+
+        for (uint256 i = 0; i < packets.length; i++) {
+            Packet memory packet = packets[i];
+            bytes32 requestKey = packet.request.getRequestKey();
+
+            require(
```
Comment (taobun) [inline bridges/evm/contracts/CacheBridge.sol]: Add internal function to verify result

File: bridges/evm/contracts/IBridge.sol [hunk 4]
```diff
@@ -25,11 +24,23 @@ interface IBridge {
         bytes result;
     }
 
+    /// Packet helps to combine RequestPacket and ResponsePacket into one struct.
```
Comment (taobun) [inline bridges/evm/contracts/IBridge.sol]: Remove this struct

**Review by taobun — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/CacheBridge.sol (hunk 3 above)
Comment (prin-r) [reply bridges/evm/contracts/CacheBridge.sol]: done

**Review by taobun — APPROVED** [review]
Thanks!

**Pull request merged**